### PR TITLE
Modal and Button

### PR DESCRIPTION
**Pull request by hchamorro**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Enhanced Button stories with new props and examples.
- Added and updated Modal stories with new examples.
- Updated Button and Modal story configurations with new controls.
- Refactored Button component to support new props and confirmation modal.
- Updated ButtonProps and ModalProps interfaces with new properties.
- Refactored Modal component with new props and content structure.
- Bumped package version to 0.47.10.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Button.stories.tsx</strong><dd><code>Enhance Button stories with new props and examples.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
apps/demo/stories/Button.stories.tsx

<li>Updated <code>ButtonDemo</code> and <code>AllColorButtons</code> to accept partial <code>ButtonProps</code>.<br> <li> Added new button types and configurations.<br> <li> Introduced full-width button examples.<br> <li> Removed deprecated button color props.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-ed90dae0169c01359c7535d917b4d7dbf48f3401785e9dc325e7b512df0ba8b9">+105/-121</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Modal.stories.tsx</strong><dd><code>Add and update Modal stories with new examples.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
apps/demo/stories/Modal.stories.tsx

<li>Added <code>ModalDemo</code> component with customizable props.<br> <li> Updated modal button examples.<br> <li> Simplified modal story structure.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-434fb5896e35a71e20b6fe03b207b8395f038dad1fe3666ee2202e76abe1f407">+40/-27</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Button.config.tsx</strong><dd><code>Update Button story configuration with new controls.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
apps/demo/story-config/Button.config.tsx

<li>Renamed and updated button stories configuration.<br> <li> Added control options for button types and icons.<br> <li> Removed deprecated button configurations.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-17cce5e4c4283f420c27f844915abd7f64e9331766ab58b69dde64f762cb13ec">+36/-8</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Modal.config.tsx</strong><dd><code>Update Modal story configuration.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
apps/demo/story-config/Modal.config.tsx

<li>Updated modal demo to use <code>ModalDemo</code>.<br> <li> Simplified modal story configuration.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-945f0a43c216dc507753653b271d509c6e8a1f51433a76108a813a3e4467f418">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Button.tsx</strong><dd><code>Refactor Button component with new props and confirmation modal.</code></dd></summary>
<hr>
      
packages/ui/src/Button.tsx

<li>Refactored <code>Button</code> component to support new props and styles.<br> <li> Added <code>ConfirmationModal</code> component for button confirmation.<br> <li> Improved button styling and layout.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-dd6ffcb0c6b3ccfcb7caa22d1c354e85bc4cdb05b40029b6d437c00fd02785c9">+147/-137</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Common.ts</strong><dd><code>Update ButtonProps and ModalProps interfaces.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/src/Common.ts

<li>Updated <code>ButtonProps</code> and <code>ModalProps</code> interfaces.<br> <li> Added new properties for button and modal configurations.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-538e93b553971318d69ff6fff2ac103f295c4922e24bbe61c30897baa463bba1">+110/-38</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Modal.tsx</strong><dd><code>Refactor Modal component with new props and content structure.</code></dd></summary>
<hr>
      
packages/ui/src/Modal.tsx

<li>Refactored <code>Modal</code> component to support new props and styles.<br> <li> Added <code>ModalContent</code> component for better structure.<br> <li> Improved modal styling and layout.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-3ab16d64d26e5ea343f785756eac0fdd776a49367e2c34dbf6b99bc524a0789f">+189/-181</a></td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Bump version to 0.47.10.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/package.json

- Bumped version to 0.47.10.

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/593/files#diff-1ae5a5e8c23bd3bc31ea590a9cbe48eb9ad3bbddc0fb5732d08c04a53c8ad51d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: apps/demo/stories/Button.stories.tsx [hunk 1]
```diff
@@ -3,63 +3,67 @@ import React from "react";
...
-        <Button text="Default" onClick={() => console.info("clicked")} {...props} />
+        <Button text="Default/Primary" onClick={() => console.info("clicked")} {...props} />
       </Box>
       <Box paddingY={1}>
         <Button
-          color="primary"
-          text="Primary"
-          tooltip={{text: "Primary Button"}}
+          text="Secondary"
+          variant="secondary"
```
Comment (joshgachnang) [inline apps/demo/stories/Button.stories.tsx:21]: This seems to be rendering a primary button still.

File: packages/ui/src/Common.ts [hunk 2]
```diff
@@ -1295,27 +1295,72 @@ export interface BodyProps {
...
+   */
+  variant?: "primary" | "secondary" | "muted" | "outline";
+  /**
+   * If true, a confirmation modal will be shown before the onClick action.
+   */
+  withConfirmationModal?: boolean;
+  /**
+   * The function to call when the button is clicked.
+   */
+  onClick: () => void;
```
Comment (joshgachnang) [inline packages/ui/src/Common.ts:1363]: I think this needs to be void | Promise<void> to make the automatic loading work.

File: packages/ui/src/Common.ts [hunk 3]
```diff
@@ -1449,29 +1494,56 @@ export interface InfoTooltipButtonProps {
...
+   */
+  visible: boolean;
+  /**
+   * The function to call when the modal is dismissed.
+   */
+  onDismiss: () => void;
+  /**
+   * The function to call when the primary button is clicked.
+   */
+  primaryButtonOnClick?: (value?: any) => void;
```
Comment (joshgachnang) [inline packages/ui/src/Common.ts:1542]: Same here, probably should be void | Promise<void> and wait to dismiss if there's an async action.

**Review by joshgachnang — COMMENTED** [review]
On iOS, the modal comes up on the demo page and cannot be dismissed.

**Comment by hchamorro** [comment]
> On iOS, the modal comes up on the demo page and cannot be dismissed.

sorry pushed up still working on swipe animation it should be all good now

File: apps/demo/stories/Button.stories.tsx (hunk 1 above)
Comment (hchamorro) [reply apps/demo/stories/Button.stories.tsx:21]: sorry i'm not seeing that issue

Comment (joshgachnang) [reply apps/demo/stories/Button.stories.tsx:21]: Ope sorry, it's the Modal secondary button that's rendering as primary

Comment (hchamorro) [reply apps/demo/stories/Button.stories.tsx:21]: okay. I do not see 2 primary buttons. Meaning they are rendering different on my end.
However the secondaryButton in Model is a button variant muted.
 Do you think i should be exclusive with the prop name
`secondaryButtonText` ->`mutedButtonText`
`secondaryButtonOnClick` -> `mutedButtonOnClick`
I'm not entirely sure if that's more of less clear as a prop

**Review by joshgachnang — APPROVED** [review]
Nice, looks and works great, good job on the swipe too! Just a handful of related components that use Modal/Button that need to be updated before merge.

**Pull request merged**